testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java

```diff
@@ -39,27 +40,11 @@ public DockerImageName apply(DockerImageName dockerImageName)
         }
 
         DockerClient client = DockerClientFactory.lazyClient();
-        try {
-            client.pullImageCmd(dockerImageName.asCanonicalNameString()).exec(new PullImageResultCallback() {
-                @Override
-                public void onNext(PullResponseItem item)
-                {
-                    String progress = item.getProgress();
-                    if (progress != null) {
-                        System.err.println(padEnd(dockerImageName.asCanonicalNameString() + ":" + item.getId(), 50, ' ') + ' ' + progress);
-                    }
-                }
-            }).awaitCompletion();
-        }
-        catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            throw new RuntimeException("Could not pull image", e);
+        if (!imageExists(client, dockerImageName)) {
```

Comment:
`testcontainers` will always try to pull the image if it doesn't exist when resolving the image name, that is when `container.getDockerImageName()` gets called. I'm trying to address this in #16259.

Reply:
Maybe you could reuse the `DefaultPullPolicy` class from `testcontainers`, it's instance can be obtained from `org.testcontainers.images.PullPolicy.defaultPolicy()`

Reply:
This is solving some other issue - when the image is built locally and doesn't belong to any registry.

Reply:
It's orthogonal to your improvement in https://github.com/trinodb/trino/pull/16259.

Reply:
Thanks for clarifying! Please mark this as resolved, I can't.